src/traits.d

```diff
@@ -522,6 +546,15 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
     {
         return isFuncX(f => !f.needThis() && !f.isNested());
     }
+    if (e.ident == Id.isModule)
+    {
+        //TODO: can we handle the case when isPkgMod == PKGunknown?
```

Comment:
>  but if I have a Package, not an Import, it's hard to get to those fields...

You can get `packages` and `id` from a `Package` object as follows:
```d
Package p;  // target
Identifiers packages;
for (Dsymbol s = p.parent; s; s = s.parent)
    packages.insert(0, s.ident);
auto id = p.ident;
// If p's fully qualified name is `a.b.c`, packages = ['a', 'b'], id = 'c'
```

Usually in D, an order dependent result is just wrong. In this case, 
deterministic behavior would also be useful, because a compiler user can check whether a package name `x` is really `package.d` or not (if `__traits(isPackage, x) && __traits(isModule, x) == true`, `x` is a `package.d`).

I think the best way is to extract a new routine from `Import.load`, which returns whether a fully qualified name is either `PKGmodule` or `PKGpackage`. In previous, the test had been needed only when a module is directly loaded from the source file list in command line, or indirectly imported from other modules. Those new traits need to do the test without importing modules. Therefore the new routine would be core of new feature.